usaspending_api/etl/management/commands/load_table_from_delta.py

```diff
@@ -215,196 +367,180 @@ def handle(self, *args, **options):
-                                                    {derived_from_cols_str})
-                        """
-                        cursor.execute(tsvector_trigger_sql)
-                        self.logger.info(f"tsvector trigger for column {tsvector_name} added.")
-
-        # Read from Delta
-        df = spark.table(delta_table)
+            self.logger.info(f"{qualified_temp_table} dropped.")
+
+        is_postgres_table_partitioned = self.table_spec.get("postgres_partition_spec") is not None
+
+        # Create the temporary table. Spark's df.write automatically does this but doesn't account for
+        # the extra metadata (indexes, constraints, defaults) which CREATE TABLE X LIKE Y accounts for.
+        # If there is no qualified_postgres_table to base it on, it just relies on spark to make it and work with delta table
+        partition_clause = ""
+        storage_parameters = "WITH (autovacuum_enabled=FALSE)"
+        partitions_sql = []
+        if is_postgres_table_partitioned:
+            partition_clause = (
+                f"PARTITION BY {self.table_spec['postgres_partition_spec']['partitioning_form']}"
+                f"({', '.join(self.table_spec['postgres_partition_spec']['partition_keys'])})"
+            )
+            storage_parameters = ""
+            partitions_sql = [
+                (
+                    f"CREATE TABLE "
+                    # Below: e.g. my_tbl_temp -> my_tbl_part_temp
+                    f"{qualified_temp_table[:-len(temp_table_suffix_appendage)]}{pt['table_suffix']}{temp_table_suffix_appendage} "
+                    f"PARTITION OF {qualified_temp_table} {pt['partitioning_clause']} "
+                    f"{storage_parameters}"
+                )
+                for pt in self.table_spec["postgres_partition_spec"]["partitions"]
+            ]
+        if schema_override:
+            create_temp_sql = f"""
+                CREATE TABLE {qualified_temp_table} (
+                    {", ".join([f'{key} {val}' for key, val in schema_override.items()])}
+                ) {partition_clause} {storage_parameters}
+            """
+        elif self.qualified_postgres_table:
+            create_temp_sql = f"""
+                CREATE TABLE {qualified_temp_table} (
+                    LIKE {self.qualified_postgres_table} INCLUDING DEFAULTS INCLUDING GENERATED INCLUDING IDENTITY
+                ) {partition_clause} {storage_parameters}
+            """
+        else:
+            raise RuntimeError(
+                "Neither a postgres_table or postgres_schema_def is populated for the target "
+                "delta table in the TABLE_SPEC and no schema override is provided"
+            )
+        with db.connection.cursor() as cursor:
+            self.logger.info(f"Creating {qualified_temp_table}")
+            cursor.execute(create_temp_sql)
+            self.logger.info(f"{qualified_temp_table} created.")
+
+            if is_postgres_table_partitioned and partitions_sql:
```

Comment:
Are `is_postgres_table_partitioned` and `partitions_sql` of the type Boolean? If not I think this if statement could be made more clear.

Reply:
`is_postgres_table_partitioned` is a Boolean, whereas `partitions_sql` is a list. Agree that it would be better to be more explicit with `partitions_sql`. Updating